central/activecomponent/updater/updater_impl_test.go

```diff
@@ -826,16 +828,38 @@ func (s *acUpdaterTestSuite) TestUpdater_Update() {
 		},
 	}
 
-	for _, testCase := range testCases {
-		s.T().Run(testCase.description, func(t *testing.T) {
+	// Note: The new loop variable semantics introduced in go1.22 breaks these
+	// tests when using loop variables in closures such that the test code
+	// expects those loop variables to change on each iteration of the loop.
+	// The problem stems from what a closure captures, or "sees", loop
+	// variables on each iteration. Before go1.22, loop variables had per-loop
+	// scope, which is to say that any closure inside the loop using loop
+	// variables would have the reference of the loop variable updated on each
+	// iteration. In go1.22+, loop variables have per-iteration scope, which,
+	// for closures inside the loop, means that when a given closure first
+	// captures the variable, that value will always be used in that particular
+	// closure.
+	// To get around this, we define variables outside the loop and then set
+	// those variables to the loop variables on each iteration, then use the
+	// variables in the outer scope (relative to the loop) in the closure,
+	// effectively achieving per-loop scoped variables.
+	// References and further reading:
+	// - https://web.archive.org/web/20240905192132/https://go.dev/blog/go1.22
+	// - https://web.archive.org/web/20240905181916/https://go.dev/blog/loopvar-preview
+	// - https://web.archive.org/web/20240905191855/https://go.dev/wiki/LoopvarExperiment
+	// - https://web.archive.org/web/20240905192225/https://go.googlesource.com/proposal/+/master/design/60078-loopvar.md
+	// - https://web.archive.org/web/20240905191656/https://go101.org/blog/2024-03-01-for-loop-semantic-changes-in-go-1.22.html
+	var tc testCase
```

Comment:
I think this would be a better solution for this test:
https://github.com/stackrox/stackrox/compare/master..michael/refactor-activecomponent-updater-test